### PR TITLE
gtkradiant: init at 1.5-2022-07-31

### PR DESCRIPTION
**Pull request by astro**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/gtkradiant/default.nix [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  desktopItems = [ desktopItem ];
+
+  installPhase = ''
+    mkdir -p $out/{bin,lib}
+    cp -ar install $out/lib/gtkradiant
+
+    ln -s ../lib/gtkradiant/radiant.bin $out/bin/gtkradiant
+    ln -s ../lib/gtkradiant/{q3map2,q3map2_urt,q3data} $out/bin/
+
+    copyDesktopItems
```
Comment (sikmir) [inline pkgs/applications/misc/gtkradiant/default.nix:150]:
```suggestion
```

Comment (astro) [reply pkgs/applications/misc/gtkradiant/default.nix]: Ah, thanks!

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 189287` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gtkradiant</li>
  </ul>
</details>

File: pkgs/applications/misc/gtkradiant/default.nix [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    ln -s ${quetooPack} $out/QuetooPack
+    ln -s ${jaPack} $out/JAPack
+    ln -s ${stvefPack} $out/STVEFPack
+    ln -s ${wolfPack} $out/WolfPack
+    ln -s ${unvanquishedPack} $out/UnvanquishedPack
+    ln -s ${q1Pack} $out/Q1Pack
+  '';
+
+  description = "Level editor for idTech games";
+
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/gtkradiant/default.nix:80]:
```suggestion
```

File: pkgs/applications/misc/gtkradiant/default.nix [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    ln -s ../lib/gtkradiant/radiant.bin $out/bin/gtkradiant
+    ln -s ../lib/gtkradiant/{q3map2,q3map2_urt,q3data} $out/bin/
+
+    mkdir -p $out/share/pixmaps
+    ln -s ../../lib/gtkradiant/bitmaps/icon.png $out/share/pixmaps/gtkradiant.png
+  '';
+
+  meta = with lib; {
+    inherit description;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/gtkradiant/default.nix:161]:
```suggestion
    description = "Level editor for idTech games";
```
Is there some kind of homepage we can set?

File: pkgs/applications/misc/gtkradiant/default.nix [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+  desktopItem = makeDesktopItem {
+    name = "gtkradiant";
+    exec = "gtkradiant";
+    desktopName = "GtkRadiant";
+    comment = description;
+    categories = [ "Development" ];
+    icon = "gtkradiant";
+    # includes its own splash screen
+    startupNotify = false;
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/gtkradiant/default.nix:90]:
```suggestion
```

File: pkgs/applications/misc/gtkradiant/default.nix [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+      python
+      svn
+      copyDesktopItems
+    ];
+
+  buildInputs = [ glib libxml2 gtk2 libGLU gnome2.gtkglext ];
+
+  enableParallelBuilding = true;
+
+  desktopItems = [ desktopItem ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/gtkradiant/default.nix:147]:
```suggestion
  desktopItems = [(makeDesktopItem {
    name = "gtkradiant";
    exec = "gtkradiant";
    desktopName = "GtkRadiant";
    comment = meta.description;
    categories = [ "Development" ];
    icon = "gtkradiant";
    # includes its own splash screen
    startupNotify = false;
  })];
```

File: pkgs/applications/misc/gtkradiant/default.nix [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+      '';
+    in [
+      scons
+      pkg-config
+      python
+      svn
+      copyDesktopItems
+    ];
+
+  buildInputs = [ glib libxml2 gtk2 libGLU gnome2.gtkglext ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/gtkradiant/default.nix:143]: Do we really need to use something from gnome2?

File: pkgs/applications/misc/gtkradiant/default.nix [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    icon = "gtkradiant";
+    # includes its own splash screen
+    startupNotify = false;
+  };
+
+in
+stdenv.mkDerivation {
+  pname = "gtkradiant";
+
+  version = "1.5-2022-07-31";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/gtkradiant/default.nix:96]:
```suggestion
  version = "unstable-2022-07-31";
```

File: pkgs/applications/misc/gtkradiant/default.nix (hunk 2 above)
Comment (astro) [reply pkgs/applications/misc/gtkradiant/default.nix:80]: This `description` is used in `makeDesktopItem`.

File: pkgs/applications/misc/gtkradiant/default.nix (hunk 6 above)
Comment (astro) [reply pkgs/applications/misc/gtkradiant/default.nix:143]: The whole project is gtk2.

File: pkgs/applications/misc/gtkradiant/default.nix (hunk 5 above)
Comment (astro) [reply pkgs/applications/misc/gtkradiant/default.nix]: Ok...

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Pull request merged**